src/mpyl/artifacts/build_artifacts.py

```diff
@@ -1,85 +1,62 @@
-            self.logger.info(f"Branch '{branch}' has no remote to pull from")
-
-    def push(self, branch: str) -> None:
-        if self.artifact_repo.does_branch_exist(branch_name=branch):
-            if self.artifact_repo.get_branch != branch:
-                self.artifact_repo.checkout_branch(branch_name=branch)
-        else:
-            if self.artifact_repo.does_branch_exist(branch_name=branch, remote=True):
-                self.logger.info(f"Fetching branch '{branch}' from remote")
-                self.artifact_repo.fetch_branch(branch_name=branch)
-                self.artifact_repo.checkout_branch(branch_name=branch)
-            else:
-                self.logger.info(f"Creating new branch '{branch}'")
-                self.artifact_repo.create_branch(branch_name=branch)
-
-        shutil.rmtree(self.cache_folder, ignore_errors=True)
-
-        for artifact_path in self.get_build_artifacts_paths():
-            shutil.copytree(
-                src=artifact_path,
-                dst=self.cache_folder / artifact_path,
-                dirs_exist_ok=True,
-            )
-
-        if self.artifact_repo.has_changes():
-            self.logger.info("Committing and pushing all artifacts")
-            self.artifact_repo.stage(CACHE_FOLDER_NAME)
-            self.artifact_repo.commit(f"Add artifacts for {branch}")
-            self.artifact_repo.push()
+        with TemporaryDirectory() as tmp_repo_dir:
+            with Repository.from_clone(
+                config=self.artifact_repo_config, repo_path=Path(tmp_repo_dir)
+            ) as artifact_repo:
+                if artifact_repo.does_branch_exist(branch_name=branch):
+                    self.logger.info(f"Fetching branch '{branch}' from remote")
+                    artifact_repo.checkout_branch(branch_name=branch)
+                self.logger.info(f"Branch {branch} does not exist in remote")
+                artifact_repo.
+
+    def push(self, branch: str, file_paths: list[Path]) -> None:
+        with TemporaryDirectory() as tmp_repo_dir:
+            repo_path = Path(tmp_repo_dir)
+            with Repository.from_clone(
+                config=self.artifact_repo_config, repo_path=repo_path
+            ) as artifact_repo:
+                branch_exists = artifact_repo.does_branch_exist(branch_name=branch)
+                if branch_exists:
+                    self.logger.info(f"Fetching branch '{branch}' from remote")
+                    artifact_repo.checkout_branch(branch_name=branch)
+                else:
+                    artifact_repo.create_branch(branch_name=branch)
+
+                shutil.rmtree(repo_path, ignore_errors=True)
+                for file_path in file_paths:
+                    shutil.copytree(
+                        src=file_path,
+                        dst=repo_path / file_path.name,
+                        dirs_exist_ok=True,
+                    )
+
+                git = artifact_repo._repo.git
+                git.add(".")
```

Comment:
What's wrong with using the Repo functions itself like I did above?

Reply:
Nothing, it's better. This was just a test.